saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/assertion/SAMLAttributeValueParser.java

```diff
@@ -141,4 +143,27 @@ public static String parseAnyTypeAsString(XMLEventReader xmlEventReader) throws
         }
     }
 
+    private static String addNamespaceWhenMissing(Deque<String> definedPrefixes, XMLEventWriter writer, StartElement startElement)
```

Comment:
Does this account for other namespaces used by e.g. attributes? For example would both `a` and `b` namespaces be preserved for the following tag?
```xml
<a:x b:lang="en" xmlns:a="uri:a" xmlns:b="uri:b"/>
```

Reply:
Namespaces for attributes was not implemented correctly. I fixed it and now all necessary namespaces are added and your example is working. I added a test for it as well.

Reply:
Perfect!